### PR TITLE
Space Travelers' Hub | React & Redux Group Project

### PR DESCRIPTION
**Pull request by Miliyonayalew**
# Space Travelers' Hub | React & Redux Group Project
## Project Description
In this project, we @Miliyonayalew  and @Enekwestanley worked with data from the SpaceX API. We built a web application for a company that provides commercial and scientific space travel services. The application allows users to book rockets and join selected space missions.

## Deployment Link
[Space Traveler' Hub](https://deploy-preview-34--rad-cat-cbc90c.netlify.app/)

## Feature Description
- Create React App.
- Install [React Redux](https://react-redux.js.org/), [Redux Logger](https://www.npmjs.com/package/redux-logger) and [React Router](https://reactrouter.com/web/guides/quick-start).
- Downloaded the [free image](https://www.flaticon.com/free-icon/planet_3212567?term=space&page=1&position=19&page=1&position=19&related_id=3212567&origin=style) for the app logo.
Created routes and view components (`Rockets`, `Missions`, `My Profile`]). 
- Used <NavLink /> for the page navigation links and style active class to indicate which section/page the user is currently on (underlined active navigation link).
- Created directories for all Redux state slice files (`rockets`, `missions`, `profile`).
- Upon the first render we fetched data from the SpaceX API endpoints.
     - Rockets: https://api.spacexdata.com/v3/rockets
     - Missions: https://api.spacexdata.com/v3/missions
- Used `useSelector()` Redux Hook to select the state slices and render lists of rockets and missions in corresponding routes.
- Styled the whole application by using `CSS3`.
- Rendered a list of rockets (as per design). For the image of a rocket used the first image in the array of flickr_images.
- Rendered the missions' data (as per design).
- Implemented My Profile page using the `filter `method to render a list of reserved rockets and missions joined depending on their status.
- Wrote unit tests with React Testing Library.
- Tested our `Rockets`, `Missions ``My Profile `and `Navbar `Components
- Apply React best practices and language style guides in code.
- Used store, actions, and reducers in React.
- Performed a code review for a team member.

## Additional Project Features
- Enhanced the My Profile section by adding a placeholder message when the "My Missions" or "My Rockets" lists are empty (`No missions joined` or `No rockets reserved`).
- Enhanced the My Profile section by adding the `Cancel reservation` and `Leave Mission` buttons to the list here. By clicking them we dispatch the actions we have already used in the main Rockets and Missions sections.

### CODE REVIEW

**Review by Ismailco — CHANGES_REQUESTED** [review]
#### Hi @Miliyonayalew, @Enekwestanley, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
You did a great job, keep up the good work 👍🏼

giphy


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by Bondok6 — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

## Hi @Miliyonayalew and @Enekwestanley 🙋‍♂️, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

TO HIGHLIGHT:

- Great job implementing the changes provided by the previous reviewer ✅
- Your project is professional ✅
- Your repo is professional ✅
- Well structured files ✅
- Professional README file 💯
- All tests are passed ✅
- No Liner errors ✅


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 




Cheers and Happy coding!👏👏👏

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/reportWebVitals.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const reportWebVitals = (onPerfEntry) => {
```
[the remark this answers is not in the document]
Comment (Miliyonayalew) [reply src/reportWebVitals.js]: Hi @Ismailco Thank you for your feedback and valuable comments, Happy Coding 😄

**Comment by Miliyonayalew** [comment]
Hello, @Bondok6 Thank you for the review , Happy Coding 😄

**Pull request merged**